File: assignments/SaveDataProject/app/src/main/java/com/example/savedataproject/NamesViewModel.java
package com.example.savedataproject;

import androidx.lifecycle.ViewModel;

import java.util.ArrayList;
import java.util.List;

public class NamesViewModel extends ViewModel {

    private final List<String> savedNames = new ArrayList<>();

    public void addName(String name) {
        savedNames.add(name);
    }

    public List<String> getSavedNames() {
        return savedNames;
    }
}

File: assignments/SaveDataProject2/app/src/main/java/com/example/savedataproject2/NamesViewModel.kt
// com.example.savedataproject2.NamesViewModel
import androidx.lifecycle.ViewModel
import androidx.databinding.ObservableField
import java.util.ArrayList

class NamesViewModel : ViewModel() {

    val names = ArrayList<String>()
    val nameInput = ObservableField<String>("")

    fun onSaveClick() {
        val enteredName = nameInput.get()?.trim() ?: ""
        if (enteredName.isNotEmpty()) {
            names.add(enteredName)
            nameInput.set("")
        }
    }

    fun getFormattedNames(): String {
        return names.joinToString("\n")
    }
}
